src/Broadway/EventStore/InMemoryEventStore.php

```diff
@@ -53,6 +56,7 @@ public function append($id, DomainEventStreamInterface $eventStream)
             $this->assertPlayhead($this->events[$id], $playhead);
 
             $this->events[$id][$playhead] = $event;
+            $this->allEvents[] = $event;
```

Comment:
I don't think you'd be able to get all events in order that way. You'd get all events in order *per aggregate root*. So if I wanted all events from the beginning of time, I'd get them all in clusters of individual aggregate roots. By adding `allEvents` we can get them in order that they were actually added.

Other two options:

 * Do some sort of deep sorting thing based on the time the events were added.
 * Rewrite the code that is aggregate root specific to read through `allEvents` each time rather than having them hashed like this.

Also possible I'm completely missing something obvious. :) But this was the rationale for adding this extra property.